src/index_update_db.js

```
@@ -2,60 +2,177 @@
+                        accountDetails, fullAdminRoleName) {
+  return new Promise((resolve, reject) => {
+    if(dbAwsAccount.account_type.toLowerCase() != 'craws') {
+      console.log("Account type is not craws. Hence skipping.");
+      resolve(true);
+    } else {
+      mcawsDbObj.Role(roleResp => {
+        roleResp.findOne({
+          where: drAdminRole
+        })
+        .then(roleData => {
+          if(roleData) {
+            return roleData;
+          } else {
+            console.log("dradmin role name provided does not exists");
+            reject("dradmin role name error");
+          }
+        })
+        .then(roleDataResp => {
+          mcawsDbObj.AwsAccountAdminRolesRoleAdminAwsAccounts(accountRole =>
+            accountRole.findOne({
+              where: {
+                awsaccount_adminroles: accountDetails.id,
+                role_adminawsaccounts: roleDataResp.dataValues.id
+              }
+            })
+            .then(accountRoleResp => {
+              if(accountRoleResp) {
+                console.log("Role already added to awsaccount");
+                return accountRoleResp;
+              } else return accountRole.create({
+                awsaccount_adminroles: accountDetails.id,
+                role_adminawsaccounts: roleDataResp.dataValues.id
+              })
+            })
+            .then(accountRoleData => {
+              console.log("Role added to awsaccount");
+              console.log(accountRoleData);
+            })
+          );
+          mcawsDbObj.AwsIamRole(iamRole =>
+            iamRole.findOne({
+              where: {
+                account: accountDetails.id,
+                name: fullAdminRoleName
+              }
+            })
+            .then(iamRoleResp => {
+              if(iamRoleResp) {
+                return iamRoleResp;
+              } else {
+                console.log("Target role does not exists");
+                reject("Target role does not exists");
```

Do we want to fail the step if the IAM role is not found?